noxfile.py

```diff
@@ -16,21 +16,41 @@ def unit(session):
     session.install("-e", ".[all,dev]", silent=False)
     if PYBOP_SCHEDULED:
         session.run("pip", "install", f"pybamm=={PYBAMM_VERSION}", silent=False)
-    session.run("pytest", "--unit")
+    session.run("pytest", "--unit", "-n", "auto")
 
 
 @nox.session
 def coverage(session):
     session.install("-e", ".[all,dev]", silent=False)
     if PYBOP_SCHEDULED:
         session.run("pip", "install", f"pybamm=={PYBAMM_VERSION}", silent=False)
+    session.run(
+        "pytest", "--unit", "--cov", "--cov-append", "--cov-report=xml", "-n", "auto"
+    )
     session.run(
         "pytest",
-        "--unit",
-        "--examples",
+        "--integration",
         "--cov",
+        "--cov-append",
         "--cov-report=xml",
+        "-n",
+        "auto",
     )
+    session.run("pytest", "--plots", "--cov", "--cov-append", "--cov-report=xml")
+
+
+@nox.session
+def integration(session):
+    session.run_always("pip", "install", "-e", ".[all,dev]")
+    session.install("pytest", "pytest-mock")
+    session.run("pytest", "--integration", "-n", "auto")
+
+
+@nox.session
+def examples(session):
+    session.run_always("pip", "install", "-e", ".[all,dev]")
+    session.install("pytest", "pytest-mock")
```

Comment:
```suggestion
```

`pytest`, `pytest-mock`, and related dependencies are already a part of `[dev]`, they can be skipped :) this applies to any other places where I missed this line.